### ISSUE TITLE
fork: Function not implemented

### ISSUE THREAD

**Issue by Zxiaoyu95** [lsds/sgx-lkl#393]
This issue occurs when I run my application with a docker image I built. I think whether my applications to run inside of SGX enclaves are not single-process? for example, the Hadoop application.

Do you know the solution to the problem? If you have good Suggestions, do you recommend how I implement fork in this project to support my application?

Looking forward to your reply! Thank you!

**Comment by davidchisnall** [comment]
Fork, in the general case, is impossible osssible to support in SGX. We do not have control over page tables and so cannot implement copy on write mappings. 

We can potentially get enough of clone working to support multiple processes created via vfork. There is a lot of work still to do there. First, we need to address #187 so that we can make the kernel aware of all of the memory mappings within a process. Then we would need to move our libc code to have a more conventional startup so that execve could work.

I would like to get there, but it’s currently a low priority. The second step will be easier once the layering cleanup is done. I’d welcome any help towards #187. If you think you might work on this, please test on top of the oe_port branch, which enables the clone system call.
